lib/util/pkcs11t.h

```diff
@@ -882,6 +882,8 @@ typedef CK_ULONG CK_MECHANISM_TYPE;
 #define CKM_AES_GCM 0x00001087
 #define CKM_AES_CCM 0x00001088
 #define CKM_AES_CTS 0x00001089
+#define CKM_AES_CMAC_GENERAL 0x0000108A
```

Comment:
May be just include a comment about which PKCS #11 spec this value came from.

Reply:
Do you have a way of finding old PKCS#11 specs so I can tell where it was introduced? The [website](http://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/pkcs11-base-v2.40.html) I've been using seems to drop them. 

Also, I noticed that the HTML version of the specs for v2.40 have the appendix with constants struck through (and point to the header file versions instead, which have different numbers). The upcoming v3 draft doesn't, and the numbers don't match the header file. 

That's the correct source for the constants, correct? Otherwise `CKM_AES_CMAC` conflicts with `CKM_AES_CTS`.

Reply:
The pkcs11 v2.40 headers are consistent with the v3.0 headers. In PKCS #11 v3.0, we removed the definitions from the spec to prevent the issues when the header and the spec doesn't match. The v2.40 errata 1 should have the correct number. Go ahead and put /* PKCS #11 v2.40 errata 1 */ in the comment and use the numbers below:
published/2-40-errata-1/pkcs11t.h:#define CKM_AES_CMAC                   0x0000108AUL
published/2-40-errata-1/pkcs11t.h:#define CKM_AES_CMAC_GENERAL           0x0000108BUL
working/3-00-current/pkcs11t.h:#define CKM_AES_CMAC                   0x0000108AUL
working/3-00-current/pkcs11t.h:#define CKM_AES_CMAC_GENERAL           0x0000108BUL
working/3-00-wd-01/pkcs11t.h:#define CKM_AES_CMAC                   0x0000108AUL
working/3-00-wd-01/pkcs11t.h:#define CKM_AES_CMAC_GENERAL           0x0000108BUL
working/identifier_db/old_dump:#define CKM_AES_CMAC         0x0000108aUL /* mechanisms - spec */
working/identifier_db/old_dump:#define CKM_AES_CMAC_GENERAL 0x0000108bUL /* mechanisms - spec */
working/identifier_db/raw_ids.db:CKM_AES_CMAC,0x0000108a,mechanisms,spec
working/identifier_db/raw_ids.db:CKM_AES_CMAC_GENERAL,0x0000108b,mechanisms,spec
[rrelyea@rrelyea-laptop pkcs11]$

Reply:
Ah, I don't have a great way to link this, but at this link: http://docs.oasis-open.org/pkcs11/pkcs11-curr/v3.0/pkcs11-curr-v3.0.html -- I still see under "B.3 Key derivation functions":

```c
#define CKM_AES_CMAC_GENERAL              0x00001089
#define CKM_AES_CMAC                      0x0000108A
```

Which I *think* is the old version of the constants. Perhaps this has been updated since 29 May 2019 and just hasn't been republished. :)